Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp

```diff
@@ -50,6 +51,19 @@ bool isValid(const Item& item)
     });
 }
 
+template<class T>
+inline static std::optional<RenderingResourceIdentifier> applyFilteredImageBufferItem(GraphicsContext& context, const ResourceHeap& resourceHeap, const T& item)
+{
+    RELEASE_ASSERT(item.sourceImageIdentifier().has_value());
+    auto resourceIdentifier = item.sourceImageIdentifier().value();
+    if (auto* sourceImage = resourceHeap.getImageBuffer(resourceIdentifier)) {
+        FilterResults results;
+        item.apply(context, sourceImage, results);
```

Comment:
I think allocating new `FilterResults` for every `Filter::apply()` is expensive if it is an `SVGFilter`. Should we cache the results like what we do in `RemoteDisplayListRecorder::drawFilteredImageBuffer()`?

Reply:
Ah, good suggestion. I was already wdnering about that, but didn't follow it - as I have other tasks pending :-)
Filed https://bugs.webkit.org/show_bug.cgi?id=269905, to take care of that. Would like to land this now, to unblock further Skia/CPU+WPE work.